src/realm-settings/NewAttributeSettings.tsx

```diff
@@ -1,23 +1,205 @@
 
-export default function NewAttributeSettings() {
+type UserProfileAttributeType = UserProfileAttribute &
+  AttributeRequired &
+  Permission;
+
+type AttributeRequired = {
+  roles: string[];
+  scopeRequired: string[];
+  enabledWhen: boolean;
+  requiredWhen: boolean;
+};
+
+type Permission = {
+  view: PermissionView[];
+  edit: PermissionEdit[];
+};
+
+type PermissionView = [
+  {
+    adminView: boolean;
+    userView: boolean;
+  }
+];
+
+type PermissionEdit = [
+  {
+    adminEdit: boolean;
+    userEdit: boolean;
+  }
+];
+
+const CreateAttributeFormContent = ({
+  save,
+}: {
+  save: (profileConfig: UserProfileConfig) => void;
+}) => {
   const { t } = useTranslation("realm-settings");
+  const form = useFormContext();
+  const history = useHistory();
+  const { realm } = useRealm();
 
   return (
-    <PageSection variant="light">
-      <FormAccess
-        onSubmit={() => console.log("TODO handle submit")}
-        isHorizontal
-        role="view-realm"
-        className="pf-u-mt-lg"
+    <UserProfileProvider>
+      <ScrollForm
+        sections={[
+          t("generalSettings"),
+          t("permission"),
+          t("validations"),
+          t("annotations"),
+        ]}
       >
-        <p>{t("createAttribute")}</p>
-      </FormAccess>
-    </PageSection>
+        <AttributeGeneralSettings form={form} />
+        <AttributePermission form={form} />
+        <AttributeValidations />
+        <AttributeAnnotations form={form} />
+      </ScrollForm>
+      <Form onSubmit={form.handleSubmit(save)}>
+        <ActionGroup className="keycloak__form_actions">
+          <Button
+            variant="primary"
+            type="submit"
+            data-testid="attribute-create"
+          >
+            {t("common:create")}
+          </Button>
+          <Button
+            variant="link"
+            onClick={() =>
+              history.push(toUserProfile({ realm, tab: "attributes" }))
+            }
+            data-testid="attribute-cancel"
+          >
+            {t("common:cancel")}
+          </Button>
+        </ActionGroup>
+      </Form>
+    </UserProfileProvider>
+  );
+};
+
+export default function NewAttributeSettings() {
+  const { realm: realmName } = useRealm();
+  const adminClient = useAdminClient();
+  const form = useForm<UserProfileConfig>();
+  const { t } = useTranslation("realm-settings");
+  const history = useHistory();
+  const { addAlert, addError } = useAlerts();
+  const [config, setConfig] = useState<UserProfileConfig | null>(null);
+  const [clientScopes, setClientScopes] =
+    useState<ClientScopeRepresentation[]>();
+
+  useFetch(
+    () => adminClient.users.getProfile({ realm: realmName }),
+    (config) => setConfig(config),
+    []
+  );
+
+  useFetch(
+    () => adminClient.clientScopes.find(),
+    (clientScopes) => {
+      setClientScopes(clientScopes);
+    },
+    []
+  );
```

Comment:
you can combine these two:
```suggestion
  useFetch(
    () =>
      Promise.all([
        adminClient.users.getProfile({ realm: realmName }),
        adminClient.clientScopes.find(),
      ]),
    ([config, clientScopes]) => {
      setConfig(config);
      setClientScopes(clientScopes);
    },
    []
  );
```